From 01f1cb8b10273eeceb999a28f576107b63907ca4 Mon Sep 17 00:00:00 2001
From: Guilhem IMBERTON <guilhem.imberton@intel.com>
Date: Thu, 3 Oct 2013 13:20:26 +0200
Subject: Fix required modules dependencies in prebuilts

Fix dependencies for BUILD_HOST_PREBUILT and BUILD_MULTI_PREBUILT

BZ: 142768
Category: Build
Domain: AOSP
Origin: Internal
Upstream-Candidate: Yes

Signed-off-by: Guilhem IMBERTON <guilhem.imberton@intel.com>
[axel.fagerstedt@intel.com: handled conflicts]
Signed-off-by: Axel Fagerstedt <axel.fagerstedt@intel.com>

Change-Id: I1a3c920a884020f15f38cb1f4f92226e8fed242b
---
 core/multi_prebuilt.mk |   38 ++++++++++++++++++++++++--------------
 1 files changed, 24 insertions(+), 14 deletions(-)

diff --git a/core/multi_prebuilt.mk b/core/multi_prebuilt.mk
index 94fe4bd..0f03b07 100644
--- a/core/multi_prebuilt.mk
+++ b/core/multi_prebuilt.mk
@@ -28,6 +28,7 @@ prebuilt_static_java_libraries := $(LOCAL_PREBUILT_STATIC_JAVA_LIBRARIES)
 prebuilt_is_host := $(LOCAL_IS_HOST_MODULE)
 prebuilt_module_tags := $(LOCAL_MODULE_TAGS)
 prebuilt_strip_module := $(LOCAL_STRIP_MODULE)
+prebuilt_required_modules := $(LOCAL_REQUIRED_MODULES)
 
 
 ifndef multi_prebuilt_once
@@ -37,12 +38,14 @@ multi_prebuilt_once := true
 # $(2): IS_HOST_MODULE
 # $(3): MODULE_CLASS
 # $(4): MODULE_TAGS
-# $(5): OVERRIDE_BUILT_MODULE_PATH
-# $(6): UNINSTALLABLE_MODULE
-# $(7): BUILT_MODULE_STEM
-# $(8): LOCAL_STRIP_MODULE
-# $(9): LOCAL_MODULE_PATH
-# $(10): LOCAL_CERTIFICATE
+# $(5): LOCAL_REQUIRED_MODULES
+# $(6): OVERRIDE_BUILT_MODULE_PATH
+# $(7): UNINSTALLABLE_MODULE
+# $(8): BUILT_MODULE_STEM
+# $(9): LOCAL_STRIP_MODULE
+# $(10): LOCAL_MODULE_PATH
+# $(11): LOCAL_CERTIFICATE
+
 #
 # Elements in the file list may be bare filenames,
 # or of the form "<modulename>:<filename>".
@@ -57,8 +60,9 @@ $(foreach t,$(1), \
   $(eval LOCAL_IS_HOST_MODULE := $(2)) \
   $(eval LOCAL_MODULE_CLASS := $(3)) \
   $(eval LOCAL_MODULE_TAGS := $(4)) \
-  $(eval OVERRIDE_BUILT_MODULE_PATH := $(5)) \
-  $(eval LOCAL_UNINSTALLABLE_MODULE := $(6)) \
+  $(eval LOCAL_REQUIRED_MODULES := $(5)) \
+  $(eval OVERRIDE_BUILT_MODULE_PATH := $(6)) \
+  $(eval LOCAL_UNINSTALLABLE_MODULE := $(7)) \
   $(eval tw := $(subst :, ,$(strip $(t)))) \
   $(if $(word 3,$(tw)),$(error $(LOCAL_PATH): Bad prebuilt filename '$(t)')) \
   $(if $(word 2,$(tw)), \
@@ -68,8 +72,8 @@ $(foreach t,$(1), \
     $(eval LOCAL_MODULE := $(basename $(notdir $(t)))) \
     $(eval LOCAL_SRC_FILES := $(t)) \
    ) \
-  $(if $(7), \
-    $(eval LOCAL_BUILT_MODULE_STEM := $(7)) \
+  $(if $(8), \
+    $(eval LOCAL_BUILT_MODULE_STEM := $(8)) \
    , \
     $(eval LOCAL_BUILT_MODULE_STEM := $(notdir $(LOCAL_SRC_FILES))) \
    ) \
@@ -78,12 +82,12 @@ $(foreach t,$(1), \
     $(eval LOCAL_MODULE_SUFFIX := $(suffix $(LOCAL_SRC_FILES))) \
    ) \
   $(if $(filter user,$(TARGET_BUILD_VARIANT)), \
-    $(eval LOCAL_STRIP_MODULE := $(8))) \
+    $(eval LOCAL_STRIP_MODULE := $(9))) \
   $(if $(9), \
-    $(eval LOCAL_MODULE_PATH := $(9)) \
+    $(eval LOCAL_MODULE_PATH := $(10)) \
    ) \
   $(if $(10), \
-    $(eval LOCAL_CERTIFICATE := $(strip $(10))) \
+    $(eval LOCAL_CERTIFICATE := $(strip $(11))) \
    ) \
   $(eval include $(BUILD_PREBUILT)) \
  )
@@ -97,6 +101,7 @@ $(call auto-prebuilt-boilerplate, \
     $(prebuilt_is_host), \
     STATIC_LIBRARIES, \
     $(prebuilt_module_tags), \
+    $(prebuilt_required_modules), \
     , \
     true)
 
@@ -105,6 +110,7 @@ $(call auto-prebuilt-boilerplate, \
     $(prebuilt_is_host), \
     SHARED_LIBRARIES, \
     $(prebuilt_module_tags), \
+    $(prebuilt_required_modules), \
     $($(if $(prebuilt_is_host),HOST,TARGET)_OUT_INTERMEDIATE_LIBRARIES), \
     , \
     , \
@@ -114,13 +120,15 @@ $(call auto-prebuilt-boilerplate, \
     $(prebuilt_executables), \
     $(prebuilt_is_host), \
     EXECUTABLES, \
-    $(prebuilt_module_tags))
+    $(prebuilt_module_tags), \
+    $(prebuilt_required_modules))
 
 $(call auto-prebuilt-boilerplate, \
     $(prebuilt_java_libraries), \
     $(prebuilt_is_host), \
     JAVA_LIBRARIES, \
     $(prebuilt_module_tags), \
+    $(prebuilt_required_modules), \
     , \
     , \
     javalib.jar)
@@ -130,6 +138,7 @@ $(call auto-prebuilt-boilerplate, \
     $(prebuilt_is_host), \
     JAVA_LIBRARIES, \
     $(prebuilt_module_tags), \
+    $(prebuilt_required_modules), \
     , \
     true, \
     javalib.jar)
@@ -141,3 +150,4 @@ prebuilt_java_libraries :=
 prebuilt_static_java_libraries :=
 prebuilt_is_host :=
 prebuilt_module_tags :=
+prebuilt_required_modules :=
-- 
1.7.4.1

